dashboard/app/controllers/authentication_options_controller.rb

```diff
@@ -11,8 +11,28 @@ def connect
   def disconnect
     return head(:bad_request) unless current_user&.migrated?
     auth_option = current_user.authentication_options.find(params[:id])
-    auth_option.destroy if auth_option.present?
 
-    return head(:no_content)
+    if auth_option.present?
+      auth_option.destroy
+
+      # Replace primary_contact_info if we just destroyed it
+      if current_user.primary_contact_info_id == auth_option.id
+        replacement = current_user.authentication_options.find_by hashed_email: auth_option.hashed_email
+        replacement ||= email_option_from_deleted_option auth_option
+        current_user.update primary_contact_info: replacement
```

Comment:
Do we care if this update fails?

Reply:
I suppose we do.  Actually, perhaps I should wrap everything from `auth_option.destroy` to this update in a transaction; seems like we don't want to modify anything if we fail this late in the process.